src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,370 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Reflection;
+using System.Reflection.Emit;
+
+// This HttpHandlerDiagnosticListener class is applicable only for .NET 4.5/4.6, and not for .NET core.
+// If you are making these changes, please test your changes manually via custom test applications.
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// A HttpHandlerDiagnosticListener is a DiagnosticListener for .NET 4.5/4.6 where HttpClient
+    /// doesn't have a DiagnosticListener built-in. This class is not used for .NET Core because
+    /// HttpClient in .NET Core already emits DiagnosticSource events. This class compensates for
+    /// that in .NET 4.5/4.6. HttpHandlerDiagnosticListener has no public constructor, but it has a
+    /// singleton style Initialize method to control its creation.
+    /// 
+    /// To use this, the application just needs to call HttpHandlerDiagnosticListener.Initialize(), and this
```

Comment:
As I mentioned elsewhere, an application cannot call Initialize because it's private. Also, please use <see cref="..." /> when referring to methods/types/etc.